<!-- Typing SVG by DenverCoder1 - https://github.com/DenverCoder1/readme-typing-svg -->
<p align="center">
  <a href="https://github.com/DenverCoder1/readme-typing-svg"><img src="https://readme-typing-svg.herokuapp.com/?lines=%20Welcome%20to%20Transcriptomics%20Lab%20;A%20diverse%20group%20of%20enthusiasts;Working%20on%20transcript%2D%27omics%27%20&center=true&width=440&height=45&color=f75c7e&vCenter=true&size=24"></a>
</p>
<h3 align="center">
  <img src="https://user-images.githubusercontent.com/88287648/130135203-b6c055e5-0839-4b64-baf2-86b55200130e.gif" width="500" height="250">
</h3>

<h2 align="center"> 
  Hello!!<img src="https://media.giphy.com/media/hvRJCLFzcasrR4ia7z/giphy.gif" width="28">
  Welcome to Transcriptomics Lab- Where we do all things RNA:mechanical_arm: :tada: </h2> 
  :man_scientist: :woman_scientist: We are research enthusiasts from diverse backgrounds in life sciences:scientist: :man_scientist:  .In our lab we study transcriptome to provide insight into how genes work:microbe: . This ReadMe provides complete analysis of an RNA-seq experiment profiling Drosophila cells after the depletion of a regulatory gene called Pasilla (Ps).

Enjoy!üåü 

# <p align = "center"> <img src = "https://www.pngkey.com/png/full/399-3992475_bonhomme-loupe-png-transparent-background-person-with-magnifying.png" width = "70" height = "70" /> </a>  Summary </p>

***A SNEAK PEEK INTO OUR LAB***

Pasilla gene üß¨ encodes a set of proteins that are most similar to those found in humans Nova-1 and Nova-2 are the names of two satellites. Nova-1 and Nova-2 are nuclear RNA-binding proteins that are normally expressed in the CNS and regulate splicing directly.There are numerous applications for RNA sequencing data with a reference genome, and there is no optimal pipeline for all cases. We will go over all  the major steps in Reference based RNA-seq data analysis üíª, such as quality control (FastQC, MutiQC, Cutadapt), reads alignment (RNA STAR, IGV), gene and transcript quantification (HT-Seq count), differential gene expression (DEseq2), functional profiling (goseq) and advanced analysis. Using this [tutorial](https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html#introduction), we were able to do differnetial  gene expression analysis besides identifying the genes and pathways regulated by the Pasilla gene as they were  affected by its depletion. To know more about our adventures ü§Ø with the analysis process, don'y hesitate to check the following hyperlinks üî•üîó. You will explore a lot about the field of transcriptomics in the next minutes üëÄ 


## <p align = "center"> üìå Our Workflow üìù  </p>
<h3 align="center">
<img src= "https://user-images.githubusercontent.com/88304342/130128293-3c8839a6-48ae-4633-b6dc-162a6afd06e8.png" width="800" height="600">
</h3>

<h1 align="center"> üìÉ Datasets and Results üìà </h1>

*Click on each image to know the detailed procedure to conduct the analysis*
<p align="center">
       <a href = "https://docs.google.com/document/d/1d4vWIoz1llimFA5zCjPaglrZrBX8Co6gbt9WcvK6l1Y/edit">
          <kbd> <img src="https://user-images.githubusercontent.com/88287648/130311901-89251a88-70b5-4004-b3b5-22f2cc77677f.png" width = "300" height ="250"> </kbd>
     </a>
  <a href="https://github.com/rana-salah/Transcriptomics/tree/Quality_Control">
      <kbd> <img src="https://user-images.githubusercontent.com/88287648/130313730-7a244afd-4687-4d04-ab7f-9e7e5d034967.png" width = "300" height = "250"> </kbd>
    </a>
    <br>
  
  <a href="https://github.com/rana-salah/Transcriptomics/tree/Mapping">
    <kbd> <img src="https://user-images.githubusercontent.com/88287648/130312527-3edcaba1-29d6-4dab-b8b2-2b8502befd68.png" width = "300" height = "250"> </kbd>
 </a>
  <br>
  <a href="https://github.com/rana-salah/Transcriptomics/tree/DE-analysis">
      <kbd> <img src="https://user-images.githubusercontent.com/88287648/130313813-34db9ff0-18e4-4f0e-a27e-0a5e1c62b222.png" width = "300" height ="250"> </kbd>
  </a>
  <a href="https://github.com/rana-salah/Transcriptomics/tree/visualization">
      <kbd> <img src="https://user-images.githubusercontent.com/88287648/130313843-8e79cbd8-9fe3-4940-99d2-784931f5fd3b.png" width = "300" height ="250"> </kbd>
  </a>
    <br>
  <a href="https://github.com/rana-salah/Transcriptomics/tree/Functional-Analysis-(GO-and-KEGG)">
    <kbd> <img src="https://user-images.githubusercontent.com/88287648/130313044-a3d0d4c5-112a-4c39-a19b-f0a9fbfab2a0.png" width = "300" height ="250"> </kbd> 

  </a>
   
      
</p>
Finally, after finishing our analysis, our team decided to start an initiative to help all the members in learning from each other and in developing the biggest set of skills during this stage! Hence, we organized a training on Friday, where we had a workshop for each step in the workflow. The highly passionate members who volunteered to give the workshops are highlighted in the contributions list. In this workshop, the moderators explained the purpose of doing each step in the tutorial and how it can be benefitial for the analysis beside highlighting some improvement points. Also, they did the anlysis process practically to help the other members follow their steps. At the end, there was a troubleshooting and a Q&A session. 

Getting to the end of our work, are excited to meet our team members?!! üòçü•≥ü•≥


# <p align = "center"> üë©‚Äçüî¨ Team Members üë®‚Äçüî¨ </p>

</p>
<h3 align="center">
<img src = "https://user-images.githubusercontent.com/67822771/130310781-f2c6ecf2-4c51-4b6e-a22a-5d79a28d83fb.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/88287648/130283020-9a0a12b4-f51e-4079-92c5-390b33ef3b8d.gif" width = "500" heigth = "200"> 
<img src = "https://user-images.githubusercontent.com/67822771/130310808-8c67a1f0-46b1-4568-b594-425a95cc789c.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130310818-c91bf2f0-921b-4833-9caf-1b68774e901c.gif" width = "500" heigth = "200"> 
<img src = "https://user-images.githubusercontent.com/67822771/130310932-50ac9646-1804-4f84-bc31-67ef491a34e9.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130310947-8ffca48e-e53a-4468-a21e-00555b7f0e60.gif" width = "500" heigth = "200"> 
<img src = "https://user-images.githubusercontent.com/67822771/130310849-6063c2f9-41f1-457b-8d3d-493ed4b6cf9e.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130310853-84e38542-67a1-454c-aa46-e0938a805edc.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130310984-3475d21c-a5e0-4159-b9e3-ed33187b7b19.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130310873-56dd33b9-8cfe-4259-8885-9a2d17705065.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130313891-6a1a2c0c-d458-4f32-89d9-de7bebf63e0d.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130313898-e940148e-df40-4465-9ed1-cf28c3874e80.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130310998-5d2766d6-99e5-4ee3-b932-08bae863cdf7.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130311001-7f77b193-6563-4d0e-bff5-c9ab9b3ccbeb.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130311010-f7e06905-edab-438b-bde8-9e6f38ebb90d.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130311013-f555b172-9544-4ae5-85f5-0ad136e03d3a.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130311057-503fa8b2-92c1-4380-8988-e9c4466a3c56.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130311063-0b93298c-37f4-4145-9b2e-9ca3805526e1.gif" width = "500" heigth = "200">
<img src = "https://user-images.githubusercontent.com/67822771/130311067-04a6646b-4679-47e5-972d-d92c619c7243.gif" width = "500" heigth = "200"> <img src = "https://user-images.githubusercontent.com/67822771/130311072-0c625935-4523-4585-8b10-f5f509e96292.gif" width = "500" heigth = "200">
  
## <p align = "center"> üë©‚Äçüíª Contributions üë®‚Äçüíª </p>


| Team Sub-groups | Specific Task | Contributors  | Slack IDs |
| :-------------: | :-----------: | :------------:| :-------: |
| Quality Control | Sub-samples   | Yasmeen & Eman | @Sam & @Eman|
| Quality Control | Full datasets | Bandana & Jaspreet| @Bandana & @Jaspreet|
| Quality Control | Markdown Documnetation | Jaspreet, Bandana, Eman| @Eman, @Bandana & @Jaspreet |
|     Mapping     | Inspection of Mapping Results | Yasmeen, Bandana, Dawoud, Nirvana | @Sam, @Bandana, @Dawoud, @Nirvana |
|     Mapping     | Counting the number of reads per annotated gene | Yasmeen, Saket, Johny | @Sam, @Saket, @Johny |
|     Mapping     | Estimation of strandness | Eman, Saket, Johny | @Saket, @Johny, @Eman|
|     Mapping     | Counting reads per genes | Ankita, Favour, Nirvana | @Anku., @Nirvana, @OYEFAVOUR |
|     Mapping     | Markdown Documentation | Eman, Yasmeen, Dawoud, Johny| @Sam & @Eman, @Dawoud, @Johny |
| Differential Gene Expression Analysis | Identification of the differentially expressed features | Rana, Utkarsha, Osama, Nikita, Chigozie, Yasmeen, Johny, Shruti| @RanaSalah, @-Utkarsha12-, @Osama, @Nikita2Chimera, @GozieNkwocha, @Sam, @Johny, @ShrutiG |
| Differential Gene Expression Analysis | Extraction of annotation of differentially expressed genes | Utkarsha, Osama, Rana, Jaspreet, Nikita, Chigozie | @RanaSalah, @-Utkarsha12-, @Osama, @Nikita2Chimera, @GozieNkwocha, @Jaspreet|
| Differential Gene Expression Analysis | Markdown Documentation | Rana| @RanaSalah |
| Visualization of the DE genes' expression | Visualization of the normalized counts | Osama, Utkarsha, Rana, Ankita, TosinA, Dawoud | @RanaSalah, @-Utkarsha12-, @Osama, @Anku., @TosinA, @Dawoud|
| Visualization of the DE genes' expression | Computation & Visualization of the Z-score | Saket, Utkarsha, Osama, Rana, Ankita, TosinA, Diyar |@Saket, @RanaSalah, @-Utkarsha12-, @Osama, @Anku., @TosinA, @diyar |
| Visualization of the DE genes' expression | Markdown Documentation | Osama, Jaspreet, Utkarsha| @Osama, @Jaspreet, @-Utkarsha12-|
| Functional enrichment analysis of the DE genes | Gene Ontology analysis | Amira, TosinA, Bandana, Johny, Shruti|@Amira, @TosinA, @Bandana, @Johny, @ShrutiG |
| Functional enrichment analysis of the DE genes | KEGG pathways analysis | Amira, TosinA, Chigozie, Rana| @Amira, @TosinA, @GozieNkwocha, @RanaSalah |
| GitHUb Markdown Development | Format & Organization   | Main ReadMe: Utkarsha, Osama, Rana, Ankita, TosinA, Bandana. Quality control: Jaspreet. Mapping: Saket, Yasmeen, Johny, Dawoud. Differential Gene Expression Analysis: Rana. Visualization: Osama, Jaspreet, Utkarsha. Functional Enrichment Analysis: TosinA | @RanaSalah, @-Utkarsha12-, @Osama, @Anku., @TosinA, @Bandana, @Jaspreet, @Sam, @Saket, @Johny, @Dawoud |
| Graphical Abstract Design |   | Rana, Osama, Jaspreet, Ankita, Diyar | @RanaSalah, @Osama, @Jaspreet, @Anku., @diyar|
| Advertisement |  Writing post on transfer-market | Tosin | @TosinA|
| Training | Moderated the training workshops & presented the workflow steps practically | Quality control: Yasmeen & Jaspreet. Mapping: Saket, Yasmeen. Differential Gene Expression Analysis: Rana & Osama. Visualization: Osama. Functional Enrichment Analysis: Amira| @Sam & @Jaspreet, @Saket, @RanaSalah, @Osama, @Amira|
  
# <p align = "center"> References </p>

Trapnell, C., L. Pachter, and S. L. Salzberg, 2009 TopHat: discovering splice junctions with RNA-Seq. Bioinformatics 25: 1105‚Äì1111. https://academic.oup.com/bioinformatics/article/25/9/1105/203994
  
Levin, J. Z., M. Yassour, X. Adiconis, C. Nusbaum, D. A. Thompson et al., 2010 Comprehensive comparative analysis of strand-specific RNA sequencing methods. Nature Methods 7: 709. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/
  
Young, M. D., M. J. Wakefield, G. K. Smyth, and A. Oshlack, 2010 Gene ontology analysis for RNA-seq: accounting for selection bias. Genome Biology 11: R14. https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-2-r14
  
Marcel, M., 2011 Cutadapt removes adapter sequences from high-throughput sequencing reads. EMBnet.journal 17: http://journal.embnet.org/index.php/embnetjournal/article/view/200
  
Brooks, A. N., L. Yang, M. O. Duff, K. D. Hansen, J. W. Park et al., 2011 Conservation of an RNA regulatory map between Drosophila and mammals. Genome Research 21: 193‚Äì202. https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3032923/
  
Robinson, J. T., H. Thorvaldsd√≥ttir, W. Winckler, M. Guttman, E. S. Lander et al., 2011 Integrative genomics viewer. Nature Biotechnology 29: 24. https://www.nature.com/nbt/journal/v29/n1/abs/nbt.1754.html
  
Wang, L., S. Wang, and W. Li, 2012 RSeQC: quality control of RNA-seq experiments. Bioinformatics 28: 2184‚Äì2185. https://www.ncbi.nlm.nih.gov/pubmed/22743226
  
Dobin, A., C. A. Davis, F. Schlesinger, J. Drenkow, C. Zaleski et al., 2013 STAR: ultrafast universal RNA-seq aligner. Bioinformatics 29: 15‚Äì21. https://academic.oup.com/bioinformatics/article/29/1/15/272537
  
Liao, Y., G. K. Smyth, and W. Shi, 2013 featureCounts: an efficient general purpose program for assigning sequence reads to genomic features. Bioinformatics 30: 923‚Äì930. https://academic.oup.com/bioinformatics/article/31/2/166/2366196
  
Kim, D., G. Pertea, C. Trapnell, H. Pimentel, R. Kelley et al., 2013 TopHat2: accurate alignment of transcriptomes in the presence of insertions, deletions and gene fusions. Genome Biology 14: R36. https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36
  
Luo, W., and C. Brouwer, 2013 Pathview: an R/Bioconductor package for pathway-based data integration and visualization. Bioinformatics 29: 1830‚Äì1831. https://academic.oup.com/bioinformatics/article-abstract/29/14/1830/232698
  
Love, M. I., W. Huber, and S. Anders, 2014 Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biology 15: 550. https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8
  
Kim, D., B. Langmead, and S. L. Salzberg, 2015 HISAT: a fast spliced aligner with low memory requirements. Nature Methods 12: 357. https://www.nature.com/articles/nmeth.3317
  
Anders, S., P. T. Pyl, and W. Huber, 2015 HTSeq‚Äîa Python framework to work with high-throughput sequencing data. Bioinformatics 31: 166‚Äì169. https://academic.oup.com/bioinformatics/article/31/2/166/2366196
  
Ewels, P., M. Magnusson, S. Lundin, and M. K√§ller, 2016 MultiQC: summarize analysis results for multiple tools and samples in a single report. Bioinformatics 32: 3047‚Äì3048. https://academic.oup.com/bioinformatics/article/32/19/3047/2196507
  
Thurmond, J., J. L. Goodman, V. B. Strelets, H. Attrill, L. S. Gramates et al., 2018 FlyBase 2.0: the next generation. Nucleic Acids Research 47: D759‚ÄìD765. https://academic.oup.com/nar/article-abstract/47/D1/D759/5144957
  
Kim, D., J. M. Paggi, C. Park, C. Bennett, and S. L. Salzberg, 2019 Graph-based genome alignment and genotyping with HISAT2 and HISAT-genotype. Nature Biotechnology 37: 907‚Äì915. https://www.nature.com/articles/s41587-019-0201-4
  
Used usegalaxy.org: "The sequencing data were uploaded to the Galaxy web platform, and we used the public server at usegalaxy.org to analyze the data ( Brooks et al. 2011)."

</h3>
</p>
<h3 align="center">

  <img src="https://user-images.githubusercontent.com/88287648/130313233-545de160-c88c-48de-9742-b9bf3cd81ebb.gif" width="1000" height="300">
</h3>

