
# Visualization of Differential gene expression

**` A: `**   After adjusting p values and annotation of genes, the most differentially expressed genes were selected. They were joined by using â€œJoin two datasetsâ€ which used the input as normalized counts generated by DESeq2. The cut tool is used to extract the columns with the gene IDs and normalized counts. The differentially expressed genes are represented in terms of log<sub>2</sub>FC(fold-change) by DESeq2.(how many times the change in expression has occurred). Then, heatmap was generated using heatmap2.
## 1. Visualisation of the normalised counts
*To extract the normalized counts for the interesting genes, we join the normalized count table generated by DESeq2 with the table we just generated.*

- Extract the normalized counts of the most differentially expressed genes

## `ğŸ‘‰ Join two Datasets`  with the following parameters:

|Steps||
|:-----:|---|
|a.|:page_facing_up: â€œJoinâ€: the `Normalized counts` file (output of DESeq2)|
|b.|â€œusing columnâ€: `Column: 1`|
|c.|:page_facing_up: â€œwithâ€: `Genes with significant adj p-value & abs(FC) > 2`|
|d.|â€œand columnâ€: `Column: 1`|
|e.|â€œKeep lines of first input that do not join with second inputâ€: `No`|
|f.|â€œKeep the header linesâ€: `Yes`|

- The generated file has more columns than we need for the heatmap: mean normalized counts, log2FC and other annotation information. We need to remove the extra columns.

|![1](https://user-images.githubusercontent.com/57266535/130300036-a40d2846-8b33-4821-aa4d-575fa58e1d87.png)|[Result](https://github.com/rana-salah/Transcriptomics/blob/1c9c5f9270f1a294564f50a718cad08aae8be598/DE%20analysis%20&%20Visualization%20Results/Galaxy38-%5BJoin_two_Datasets_on_data_37_and_data_25%5D.tabular) |
|---|---|


## `ğŸ‘‰ Cut Tool:` to extract the columns with the gene IDs and normalized counts

|Steps||
|:-----:|---|
|a.|"Cut columnsâ€: `c1-c8`|
|b.|â€œDelimited byâ€: `Tab`|
|c.|:page_facing_up: â€œFromâ€: the joined dataset (output of Join two Datasets)|

- Rename the output to `Normalized counts for the most differentially expressed genes`

| ![2](https://user-images.githubusercontent.com/57266535/130301117-db901921-74f4-410c-ae0e-a41b59cf880c.png)|[Result](https://github.com/rana-salah/Transcriptomics/blob/1c9c5f9270f1a294564f50a718cad08aae8be598/DE%20analysis%20&%20Visualization%20Results/Galaxy39-%5BNormalized_counts_for_the_most_differentially_expressed_genes%5D.tabular) |
|---|---|

 ## `ğŸ‘‰ To plot the heatmap`
|Steps||
|:-----:|---|
|  a.| :page_facing_up: Input should have column headersâ€: `Normalized counts for the most differentially expressed genes` |
|  b.| Advanced - log transformationâ€/â€œData transformationâ€: `Log2(value) transform my data` |
|  c.| Enable data clusteringâ€: `Yes`|
|  d.| Labeling columns and rowsâ€: `Label columns and not rows` |
|  e.| Coloring groupsâ€: `Blue to white to red`|

### `ğŸ‘‰ Result`

<h1 align="center">

![6](https://user-images.githubusercontent.com/57266535/130248627-46327f99-9921-469e-b3a5-a7d19f737efb.png) 
</h3>

The samples are clustering by treatment, all the treated csamples can be visualised togteher and untreated are together on the heatmap.


## 2. Computation and Visualisation of Z-score
*The Z-score gives the number of standard-deviations that a value is away from the mean of all the values in the same group, here the same gene.*

Another statistical measure is to calculate the z score and represent expression in terms of deviation from mean. This is done using table compute function for normalized counts; which says how far from mean values our actual readings are (in terms of standard deviations). Note that +-3SD has around 99.6% of data. Lower expressed (downregulated) genes will have negative SD, and upregulated will have positive SD. Our samples will show a plus or minus representation. Another heatmap is plotted to see how individual samples fare.

## `ğŸ‘‰ Table Compute` with the following parameters to first subtract the mean values per row
- â€œInput Single or Multiple Tablesâ€: Single Table

|Steps||
|:-----:|---|
|a.| :page_facing_up: â€œTableâ€: `Normalized counts for the most differentially expressed genes`|
|b.|â€œType of table operationâ€: `Perform a full table operation`|
|c.|Custom expression on â€˜tableâ€™, along â€˜axisâ€™ (0 or 1)â€: `table.sub(table.mean(1), 0)`|
|d.|The table.mean(1) expression computes the mean for each row (here the genes) and table.sub(table.mean(1), 0) subtracts each value by the mean of the row (computed with table.mean(1))|

|![3](https://user-images.githubusercontent.com/57266535/130301119-8670666b-9302-4e6a-9515-a2d23120ad21.png)|[Result](https://github.com/rana-salah/Transcriptomics/blob/577eb6ad9b4c9b5e66650a67d9933e80262fe415/DE%20analysis%20&%20Visualization%20Results/Galaxy41-%5BTable_Compute_on_data_39%5D.tabular) |
|---|---|

## `ğŸ‘‰ Table Compute Tool`
- â€œInput Single or Multiple Tablesâ€: Multiple Table

|Steps||
|:-----:|---|
|a.|:page_facing_up: â€œTable1â€: `Normalized counts for the most differentially expressed genes`|
|b.|Click on: â€œInsert Tablesâ€|
|c.|â€œTable2â€: `Table compute output`|
|d.|Custom expression on `â€˜table2.div(table1.std(1),0)â€˜`|
|e.|he table1.std(1) expression computes the standard deviations of each row on the 1st table (normalized counts) and table2.div divides the values of 2nd table (previously computed) by these standard deviations.|
|f.|Rename the output to Z-scores for the most differentially expressed genes|

|![4](https://user-images.githubusercontent.com/57266535/130301120-020e9cb7-f8a0-44d1-9b3d-9472213cae80.png)|[Result](https://github.com/rana-salah/Transcriptomics/blob/577eb6ad9b4c9b5e66650a67d9933e80262fe415/DE%20analysis%20&%20Visualization%20Results/Galaxy42-%5BZ-scores_for_the_most_differentially_expressed_genes%5D.tabular) |
|---|---|

## `ğŸ‘‰ To plot the heatmap`

|Steps||
|:-----:|---|
|a.|:page_facing_up: â€œInput should have column headersâ€: `Z-scores for the most differentially expressed genes`|
|b.|â€œAdvanced - log transformationâ€/â€œData transformationâ€: `Plot the data as it is`|
|c.|â€œEnable data clusteringâ€: `Yes`|
|d.|â€œLabeling columns and rowsâ€: `Label columns and not rows`|
|e.|â€œColoring groupsâ€: `Blue to white to red`|

<h3 align="center">
    
![5](https://user-images.githubusercontent.com/57266535/130301121-8c4df88e-4724-4842-8a49-88c3655a2d45.png)
</h1>

### `ğŸ‘‰ Result`

<h3 align="center">
 
![7](https://user-images.githubusercontent.com/57266535/130248956-0a218b31-76e6-417f-8627-33a2577141b1.png)
</h1>





</h3>
The Z-score ranges from -3 standard deviations up to +3 standard deviations. When a gene is differentially expressed between two groups (here treated and untreated), the Z-scores for this gene will be (mostly) positive for the samples in one group and (mostly) negative for the samples in the other group. 
